azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureAsyncOperationPollStrategy.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.v2;
+
+import com.microsoft.rest.protocol.SerializerAdapter;
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+import rx.functions.Func1;
+
+import java.io.IOException;
+
+/**
+ * A PollStrategy type that uses the Azure-AsyncOperation header value to check the status of a long
+ * running operation.
+ */
+public final class AzureAsyncOperationPollStrategy extends PollStrategy {
+    private final String fullyQualifiedMethodName;
+    private final String operationResourceUrl;
+    private final String originalResourceUrl;
+    private final SerializerAdapter<?> serializer;
+    private Boolean pollingSucceeded;
```

Comment:
there are few methods in this class which checks `pollingSucceeded` is NULL vs TRUE vs FALSE. I wish there Nullable<T> in jdk.. :) ..

Reply:
Even `Boolean.isTrue()` which doesn't require a null comparison would be welcome.

Reply:
I was using the pollingSucceeded Boolean as a ternary state value (null meant that polling hadn't completed, true meant that it has completed successfully, false meant it had completed unsuccessfully), but I've converted it into two booleans (pollingCompleted and pollingSucceeded).